src/sonic-yang-models/sonic-acl.yang

```diff
@@ -0,0 +1,231 @@
+module sonic-acl {
+	namespace "http://github.com/Azure/sonic-acl";
+	prefix sacl;
+	yang-version 1.1;
+
+	import ietf-inet-types {
+		prefix inet;
+	}
+
+	import sonic-common {
+		prefix cmn;
+	}
+
+	import sonic-extension {
+		prefix sonic-ext;
+	}
+
+	import sonic-port {
+		prefix prt;
+	}
+
+	import sonic-mirror-session {
+		prefix sms;
+	}
+
+	organization
+		"SONiC";
+
+	contact
+		"SONiC";
+
+	description
+		"SONIC ACL";
+
+	revision 2019-05-15 {
+		description
+			"Initial revision.";
+	}
+
+	container sonic-acl {
+
+		container ACL_TABLE {
+
+			list ACL_TABLE_LIST {
+				key "aclname";
+				max-elements 1024; // Max 1K ACL tables for all platforms
+				sonic-ext:custom-validation ValidateMaxAclTable;
+
+				leaf aclname {
+					type string {
+						pattern '[a-zA-Z0-9]{1}([-a-zA-Z0-9_]{0,71})';
+						length 1..72;
+					}
+				}
+
+				leaf policy_desc {
+					type string {
+						length 1..255 {
+							error-app-tag policy-desc-invalid-length;
+						}
+					}
+				}
+
+				leaf stage {
+					type enumeration {
+						enum INGRESS;
+						enum EGRESS;
+					}
+				}
+
+				leaf type {
+					type enumeration {
+						enum MIRROR;
+						enum L2;
+						enum L3;
+						enum L3V6;
+					}
+				}
+
+				leaf-list ports {
+					type leafref {
+						path "/prt:sonic-port/prt:PORT/prt:PORT_LIST/prt:ifname";
+					}
+				}
+			}
+		}
+
+		container ACL_RULE {
+
+			list ACL_RULE_LIST { 
+				key "aclname rulename";
+				max-elements 65536; // Max 64K ACL rules for all platforms
+
+				leaf aclname {
+					type leafref {
+						path "../../../ACL_TABLE/ACL_TABLE_LIST/aclname";
+					}
+					must "(/cmn:operation/cmn:operation != 'DELETE') or " +
+						"count(current()/../../../ACL_TABLE/ACL_TABLE_LIST[aclname=current()]/ports) = 0" {
+							error-message "Ports are already bound to this rule.";
+						}
+				} 
+
+				leaf rulename {
+					type string;
+				}
+
+				leaf PRIORITY {
+					type uint16 {
+						range "1..65535"{
+							error-message "Invalid ACL rule priority.";
+					}
+					}
+				}
+
+				leaf RULE_DESCRIPTION {
+					type string;
+				}
+
+				leaf PACKET_ACTION {
```

Comment:
Should we move all custom enumeration definition in sonic-common. Easy to look and change.

Reply:
We should not be doing all the custom enumerations in one single file. As it will become a huge file, once we have all yangs and their corresponding enumerations. I would prefer a common folder in which people can create their own enumeration files.